batchDivision/batch_division.go

```diff
@@ -0,0 +1,80 @@
+package batchDivision
+
+import (
+	"github.com/codeuniversity/al-proto"
+	"math"
+	"math/rand"
+	"strconv"
+	"time"
+)
+
+func createBatches(cells []*proto.Cell, batchSize uint) map[string][]*proto.Cell {
+	dict := make(map[string][]*proto.Cell)
+
+	for i := 0; i < len(cells); i++ {
+		key := createBatchKey(cells[i], batchSize)
+
+		if val, ok := dict[key]; ok {
+			dict[key] = append(val, cells[i])
+		} else {
+			dict[key] = []*proto.Cell{cells[i]}
+		}
+	}
+	return dict
+}
+
+func createBatchKey(cell *proto.Cell, batchSize uint) string {
```

Comment:
you only ever use the Pos of the cell, so this func could just be `keyFor(pos *proto.Vector, batchSize uint)` :)